src/extensions.jl

```diff
@@ -83,10 +93,8 @@ tags[:array] = d ->
 
 # Structs
 
-isprimitive(T) = fieldcount(T) == 0 && T.size > 0
-
-structdata(x) = isprimitive(typeof(x)) ? reinterpret_(UInt8, [x]) :
-    Any[getfield(x, f) for f in fieldnames(typeof(x))]
+structdata(x) = isprimitivetype(typeof(x)) ? reinterpret_(UInt8, [x]) :
+    Any[getfield(x,f) for f in fieldnames(typeof(x)) if isdefined(x, f)]
```

Comment:
The `if isdefined(x, f)` was necessary to avoid errors.  I don't actually know what it does... So, if this leads to a bug, it would be good to have a test-case which shows this.

Reply:
What are the errors? We'd need all the fields of a struct to reconstruct it iiuc

Reply:
Just to clarify, this works with master + 1.6 but errors with 1.7?